### PR TITLE
Updated code lite to properly add LD_LIBRARY_PATH is set if libdirs is set

### PR DESCRIPTION
**Pull request by pintodragon**
Also added a change to the tree.traverse to make it generate the base virtual directories. Without this change I was getting projects that had files that were not part of a virtual directory. These files would not show up in the editor and there was no way for me to easily find them and move them to a virtual directory for them to show up.

### CODE REVIEW

File: modules/codelite/codelite_project.lua [hunk 1]
```diff
@@ -268,9 +267,21 @@
 		local pauseexec  = iif(prj.kind == "ConsoleApp", "yes", "no")
 		local isguiprogram = iif(prj.kind == "WindowedApp", "yes", "no")
 		local isenabled  = iif(cfg.flags.ExcludeFromBuild, "no", "yes")
-
-		_x(3, '<General OutputFile="%s" IntermediateDirectory="%s" Command="%s" CommandArguments="%s" UseSeparateDebugArgs="%s" DebugArguments="%s" WorkingDirectory="%s" PauseExecWhenProcTerminates="%s" IsGUIProgram="%s" IsEnabled="%s"/>',
-			targetname, objdir, command, cmdargs, useseparatedebugargs, debugargs, workingdir, pauseexec, isguiprogram, isenabled)
+    local ldPath = ''
+    
+    for _, libdir in ipairs(cfg.libdirs) do
+			ldPath = ldPath .. ";" .. project.getrelative(cfg.project, libdir)
```
Comment (pintodragon) [inline modules/codelite/codelite_project.lua]: The ; should be platform specific.

File: modules/codelite/codelite_project.lua [hunk 2]
```diff
@@ -111,7 +111,7 @@
 
 		-- TODO: ...
 	end
-
+
```
Comment (samsinsane) [inline modules/codelite/codelite_project.lua]: This can probably be reverted.

File: modules/codelite/codelite_project.lua [hunk 3]
```diff
@@ -268,9 +267,21 @@
 		local pauseexec  = iif(prj.kind == "ConsoleApp", "yes", "no")
 		local isguiprogram = iif(prj.kind == "WindowedApp", "yes", "no")
 		local isenabled  = iif(cfg.flags.ExcludeFromBuild, "no", "yes")
-
-		_x(3, '<General OutputFile="%s" IntermediateDirectory="%s" Command="%s" CommandArguments="%s" UseSeparateDebugArgs="%s" DebugArguments="%s" WorkingDirectory="%s" PauseExecWhenProcTerminates="%s" IsGUIProgram="%s" IsEnabled="%s"/>',
-			targetname, objdir, command, cmdargs, useseparatedebugargs, debugargs, workingdir, pauseexec, isguiprogram, isenabled)
+    local ldPath = ''
```
Comment (samsinsane) [inline modules/codelite/codelite_project.lua]: Somehow I managed to not notice this before, the indentation here isn't right. It looks like you're using spaces while the rest of the file is using tabs.

Comment (pintodragon) [reply modules/codelite/codelite_project.lua]: Ah, I never use tabs.

Also the LD_LIBRARY_PATH may be linux specific. So I will have to figure out how to put the correct value in there for windows.

File: modules/codelite/codelite_project.lua (hunk 2 above)
Comment (pintodragon) [reply modules/codelite/codelite_project.lua]: Adding the extra line here actually conforms to the rest of the file so I will leave it.

**Comment by pintodragon** [comment]
This should be ready to merge now.

**Comment by pintodragon** [comment]
I did and in the process ended up having the process closed. I can raise a new one if needed

**Pull request closed without merging**